app/controllers/stages_controller.rb

```diff
@@ -145,4 +145,10 @@ def find_project
   def find_stage
     @stage = @project.stages.find_by_param!(params[:id])
   end
+
+  def update_deploy_groups
+    # http://stackoverflow.com/questions/8929230/why-is-the-first-element-always-blank-in-my-rails-multi-select-using-an-embedde
+    deploy_group_ids = params[:stage].fetch(:deploy_groups, []).reject(&:empty?)
+    @stage.deploy_groups = DeployGroup.find(deploy_group_ids)
```

Comment:
assigning an association auto-persists it

On Tue, Mar 3, 2015 at 3:13 PM, Roman Sandler <notifications@github.com>
wrote:

> In app/controllers/stages_controller.rb
> <https://github.com/zendesk/samson/pull/251#discussion_r25737753>:
>
> > @@ -145,4 +145,10 @@ def find_project
> >    def find_stage
> >      @stage = @project.stages.find_by_param!(params[:id])
> >    end
> > +
> > +  def update_deploy_groups
> > +    # http://stackoverflow.com/questions/8929230/why-is-the-first-element-always-blank-in-my-rails-multi-select-using-an-embedde
> > +    deploy_group_ids = params[:stage].fetch(:deploy_groups, []).reject(&:empty?)
> > +    @stage.deploy_groups = DeployGroup.find(deploy_group_ids)
>
> how is this persisted?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/251/files#r25737753>.
>